### PR TITLE
chore: restructuring design system react component code into folders

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## **Description**

This PR restructures the existing example components within the `design-system-react` package by moving them into individual component folders. This organization update aims to prepare for the addition of `Text` and other new components, promoting better organization and modularity within the component library. This change facilitates easier navigation and scalability as new components are added.

## **Related Issues**

Fixes: N/A

## **Manual Testing Steps**

1. Run storybook locally `yarn storybook`
2. Confirm that each component still renders correctly without any errors.
3. Run `yarn build` ensure that components are still built

### **Before**

https://github.com/user-attachments/assets/e2d54899-169d-415f-b834-3da2b3de6f60

### **After**

https://github.com/user-attachments/assets/427a04ac-7ec7-42f6-bc86-50dd8d3067d0

## **Pre-merge Author Checklist**

- [x] Followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs)
- [x] Completed this PR template fully
- [x] Included tests, if applicable
- [x] Documented code changes using [JSDoc](https://jsdoc.app/) format, if applicable
- [x] Applied appropriate labels (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md))

## **Pre-merge Reviewer Checklist**

- [ ] Manually tested the PR by pulling and building the branch, running the app, and confirming the functionality of all updated components.
- [ ] Verified that this PR addresses all outlined acceptance criteria and includes necessary evidence (screenshots/recordings if applicable).

### CODE REVIEW

File: packages/design-system-react/jest.config.js [hunk 1]
```diff
@@ -14,6 +14,9 @@ module.exports = merge(baseConfig, {
   // The display name when running multiple projects
   displayName,
 
+  // Add coverage ignore patterns
+  coveragePathIgnorePatterns: ['/node_modules/', 'index.ts'],
```
Comment (georgewrmarshall) [inline packages/design-system-react/jest.config.js:18]: Ignoring index files from testing coverage

File: packages/design-system-react/jest.config.js [hunk 2]
```diff
@@ -14,6 +14,9 @@ module.exports = merge(baseConfig, {
   // The display name when running multiple projects
   displayName,
 
+  // Add coverage ignore patterns
+  coveragePathIgnorePatterns: ['index.ts'],
```
Comment (georgewrmarshall) [inline packages/design-system-react/jest.config.js:18]: Ignoring index files from test coverage as they only re-export components. We could add tests for these in the future, but will ignore them for now.

File: packages/design-system-react/src/components/button/index.ts [hunk 3]
```diff
@@ -0,0 +1,2 @@
+export { Button } from './Button';
+export type { ButtonProps } from './Button';
```
Comment (georgewrmarshall) [inline packages/design-system-react/src/components/button/index.ts:2]: Adding index file for Button

File: packages/design-system-react/src/components/index.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export { Text, TextVariant } from './text';
+export type { TextProps } from './text';
+
+export { Button } from './button';
+export type { ButtonProps } from './button';
```
Comment (georgewrmarshall) [inline packages/design-system-react/src/components/index.ts:5]: Adding root index file in components folder

File: packages/design-system-react/src/components/text/index.ts [hunk 5]
```diff
@@ -0,0 +1,2 @@
+export { Text, TextVariant } from './Text';
```
Comment (georgewrmarshall) [inline packages/design-system-react/src/components/text/index.ts:1]: Exporting Text and Text types

File: packages/design-system-react/src/index.test.ts
Comment (georgewrmarshall) [inline packages/design-system-react/src/index.test.ts:1]: Removing unnecessary index test file as we are now ignoring index files from testing coverage

**Review by brianacnguyen — APPROVED** [review]

**Pull request merged**